src/oop_ext/foundation/weak_ref.py

```diff
@@ -354,23 +354,23 @@ def GetRealObj(obj: Any) -> Any:
     return obj
 
 
-def GetWeakProxy(obj: Any) -> Any:
+def GetWeakProxy(obj: T) -> T:
     """
     :param obj: This is the object we want to get as a proxy
     :return:
         Returns the object as a proxy (if it is still not already a proxy or a weak ref, in which case the passed object
         is returned itself)
     """
     if obj is None:
-        return None
+        return None  # type:ignore[return-value]
```

Comment:
Just out of curiosity, would `return obj` work here without the `type:ignore`?

Reply:
Good call! It did work, but after your other suggestions, I changed the return type to `ProxyType[T]`, and this still requires the type:ignore.